### ISSUE TITLE
BUG: GeoDataFrame.pivot fails

### ISSUE THREAD

**Issue by m-richards** [geopandas/geopandas#2057]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of geopandas.

- [x] (optional) I have confirmed this bug exists on the master branch of geopandas.

---

#### Code Sample

```python
from geopandas import GeoDataFrame
from shapely.geometry import Point
import pandas as pd
d = GeoDataFrame({'col1': [1, 2], 'geometry': [Point(1, 2), Point(2, 1)]})
print(d)
print(pd.DataFrame(d).pivot(columns='col1'))
print(d.pivot(columns='col1')) # this crashes
#
   col1                 geometry
0     1  POINT (1.00000 2.00000)
1     2  POINT (2.00000 1.00000)
                     geometry                         
col1                        1                        2
0     POINT (1.00000 2.00000)                     None
1                        None  POINT (2.00000 1.00000)
```

#### Problem description
Pivot should work with geometries present

#### Expected Output
Same as the DataFrame case above
#### Output of ``geopandas.show_versions()``

<details>

SYSTEM INFO
-----------
python     : 3.8.10 | packaged by conda-forge | (default, May 11 2021, 06:25:23) [MSC v.1916 64 bit (AMD64)]
executable : C:\MyPrograms\Miniconda3\envs\gpd_latest_conda_forge\python.exe
machine    : Windows-10-10.0.19043-SP0

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : None
GEOS lib   : None
GDAL       : 3.3.1
GDAL data dir: None
PROJ       : 8.0.1
PROJ data dir: C:\MyPrograms\Miniconda3\envs\gpd_latest_conda_forge\Library\share\proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 0.9.0+58.g8547d7f
pandas     : 1.3.1
fiona      : 1.8.20
numpy      : 1.21.1
shapely    : 1.7.1
rtree      : 0.9.7
pyproj     : 3.1.0
matplotlib : 3.4.2
mapclassify: 2.4.3
geopy      : 2.2.0
psycopg2   : 2.9.1 (dt dec pq3 ext lo64)
geoalchemy2: None
pyarrow    : 5.0.0
pygeos     : 0.10.1
None

</details>

**Comment by jorisvandenbossche** [comment]
The problem is similar as the "duplicate column name" you were working on. The issue is that the pivoted table has a MultiIndex with "geometry" as label in the first level. But so doing

**Comment by jorisvandenbossche** [comment]
So we might need to expand the check  that you added at

https://github.com/geopandas/geopandas/blob/4536b9203b0b2ff577d1368e4677ff93a4c82164/geopandas/geodataframe.py#L124-L128

to also deal with MultiIndex columns. 

And then with https://github.com/geopandas/geopandas/pull/2055#issuecomment-894853432, `pivot` can fall back to returning a normal DataFrame.

**Comment by m-richards** [comment]
Ah that's neat this is linked, I hadn't looked into it at all, just stumbled across this bug at work today. I can probably fix this as well then.

**Comment by m-richards** [comment]
This was fixed by #2497
